src/Appwrite/Utopia/Response/Model/Message.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Appwrite\Utopia\Response\Model;
+
+use Appwrite\Utopia\Response;
+use Appwrite\Utopia\Response\Model;
+use Utopia\Database\DateTime;
+
+class Message extends Any
+{
+    public function __construct()
+    {
+        $this
+            ->addRule('$id', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Message ID.',
+                'default' => '',
+                'example' => '5e5ea5c16897e',
+            ])
+            ->addRule('providerId', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Provider ID for the message.',
+                'default' => '',
+                'example' => '5e5ea5c16897e',
+            ])
+            ->addRule('to', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Message recipients.',
+                'default' => '',
+                'array' => true,
+                'example' => ['user-1'],
+            ])
+            ->addRule('deliveryTime', [
+                'type' => self::TYPE_DATETIME,
+                'description' => 'The scheduled time for message.',
+                'required' => false,
+                'default' => DateTime::now(),
+                'example' => self::TYPE_DATETIME_EXAMPLE,
+            ])
+            ->addRule('deliveredAt', [
+                'type' => self::TYPE_DATETIME,
+                'description' => 'The time when the message was delivered.',
+                'required' => false,
+                'default' => '',
+                'example' => self::TYPE_DATETIME_EXAMPLE,
+            ])
+            ->addRule('deliveryErrors', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Delivery errors if any.',
+                'required' => false,
+                'default' => '',
+                'array' => true,
+                'example' => ['Failed to send message to target 5e5ea5c16897e: Credentials not valid.'],
+            ])
+            ->addRule('deliveredTo', [
```

Comment:
A bit misleading. My initial thought was that this will be an array of all targets.

Reply:
How about `deliveryCount`? This will be how many targets the message has currently been delivered to, updated as the worker processes the target batches

Reply:
How about `deliveredTotal` considering we use total in other places?

Reply:
Changed to ``deliveredTotal``